iquip/apps/builder.py

```diff
@@ -320,16 +324,33 @@ def __init__(self, name: str, argInfo: Dict[str, Any], parent: Optional[QWidget]
               ndecimals: The number of displayed decimals.
         """
         super().__init__(name, argInfo, parent=parent)
-        self.state = self.get_state()
-        procdesc = self.get_procdesc()
-        # TODO(AIJUH): Add features for stack widget.
-        self.stackWidget = QStackedWidget(self)
-        self.rangeWidget = _RangeScan(procdesc, self.state["RangeScan"])
+        self.state = self.stateManager()
+        procdesc = self.procdescManager()
+        self.stackedWidget = QStackedWidget(self)
+        buttonLayout = QHBoxLayout()
+        self.scanButtonGroup = QButtonGroup(self)
+        scanDict = {"NoScan": (_NoScan, "No scan"), "RangeScan": (_RangeScan, "Range")}
+        self.scanWidgetsDict = {}
+        self.scanButtonsDict = {}
+        for buttonId, (ty, (scanCls, buttonName)) in enumerate(scanDict.items()):
+            scanWidget = scanCls(procdesc, self.state[ty])
+            self.scanWidgetsDict[ty] = scanWidget
+            self.stackedWidget.addWidget(scanWidget)
+            button = QRadioButton(buttonName)
+            self.scanButtonsDict[ty] = buttonId
+            buttonLayout.addWidget(button)
+            self.scanButtonGroup.addButton(button, buttonId)
+        self.scanButtonGroup.buttonToggled.connect(self.scanTypeToggled)
+        selected = self.state["selected"]
+        selectedButton = self.scanButtonGroup.button(self.scanButtonsDict[selected])
+        selectedButton.setChecked(True)
+        # layout
         scanLayout = QVBoxLayout()
-        scanLayout.addWidget(self.rangeWidget)
+        scanLayout.addLayout(buttonLayout)
+        scanLayout.addWidget(self.stackedWidget)
         self.layout.addLayout(scanLayout)
 
-    def get_state(self) -> Dict[str, Any]:
+    def stateManager(self) -> Dict[str, Any]:
```

Comment:
First, we have usually named a method to start with a verb, except to a slot.
Also `*Manager` feels like it means a manager object.

Then, how about just `initState()` or something better?

Reply:
I also thought abot that.
I thought starting with `init` might cause problem.